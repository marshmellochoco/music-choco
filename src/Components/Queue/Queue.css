:root {
    --primary-color: #404040;
    --secondary-color: #707070;
    --thirtiary-color: #a0a0a0;
    --contrast-color: #d0d0d0;
}

.queue{
    background-color: var(--secondary-color);
    padding: 0 1rem;
    overflow-y: scroll;
}

.queueList{
    list-style-type: none;
    padding-left: 0;
}

.queueList li{
    padding: 3% 4%;
    display: grid;
    grid-template-columns: 1fr 3em 25px;
    grid-template-rows: 1fr 1fr;
}

.queueList li .album{
    font-size: 0.8em;
    grid-row: 2;
}

.queueList li .duration{
    display: flex;
    align-self: center;
    grid-row: 1/3;
    grid-column: 2;
}

.queueList li .removeButton{
    width: 25px;
    height: 25px;
    grid-row: 1/3;
    grid-column: 3;
    cursor:pointer;
    display: flex;
    align-self: center;
}

.queueList li .icon{
    width: 100%;
    height: 100%;
}

.queueList .queueItem:hover{
    background-color: var(--primary-color);
}

.queueList .queueActive{
    background-color: black;
}