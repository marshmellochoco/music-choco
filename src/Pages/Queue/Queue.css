:root {
    --primary-color: #404040;
    --secondary-color: #707070;
    --thirtiary-color: #a0a0a0;
    --contrast-color: #d0d0d0;
}

.queue {
    background-color: var(--secondary-color);
    padding: 0 1rem;
    overflow-y: scroll;
}

.queue .queueHeader {
    display: flex;
    align-items: baseline;
    align-content: flex-end;
    margin-left: auto;
}

.queue .queueHeader .controls {
    display: flex;
    margin-left: auto;
}

.queueHeader .controls .icons {
    fill: var(--contrast-color);
    width: 2em;
    height: 2em;
    margin: 5%;
}

.queueHeader .controls .active {
    color: var(--primary-color);
}


.queue .loading {
    margin: auto;
    color: #404040;
}
