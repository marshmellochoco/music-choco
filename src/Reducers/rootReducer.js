import { queueReducer } from "../Reducers/queueReducer";
import { songDataReducer } from "../Reducers/songDataReducer";
import { playerReducer } from "../Reducers/playerReducer";
import { authReducer } from "../Reducers/authReducer";

import { combineReducers } from "redux";

export const rootReducer = combineReducers({
    queueReducer,
    songDataReducer,
    playerReducer,
    authReducer,
});
