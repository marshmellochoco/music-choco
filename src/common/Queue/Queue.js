import { useDispatch, useSelector } from "react-redux";
import { setPlaying, setPlayingTrack } from "../../store/player/playerAction";
import { setQueue } from "../../store/queue/queueAction";
import QueueComponent from "./QueueComponent";

const Queue = ({ openQueue, setOpenQueue }) => {
    const dispatch = useDispatch();
    const queueData = useSelector((state) => state.queueReducer);
    const { playingTrack } = useSelector((state) => state.playerReducer);

    const handleDragDrop = (result) => {
        if (!result.destination) return;

        const q = [...queueData];
        const [s] = q.splice(result.source.index, 1);
        q.splice(result.destination.index, 0, s);
        dispatch(setQueue(q));
    };

    const setTrack = (trackData) => {
        dispatch(setPlayingTrack(trackData));
        dispatch(setPlaying(true));
    };

    return (
        <>
            {openQueue && (
                <QueueComponent
                    queueData={queueData}
                    playingTrack={playingTrack}
                    setTrack={setTrack}
                    handleDragDrop={handleDragDrop}
                    setOpenQueue={setOpenQueue}
                />
            )}
        </>
    );
};

export default Queue;
