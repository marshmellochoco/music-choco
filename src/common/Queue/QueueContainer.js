import { QueueComponent } from "./QueueComponent";
import { useDispatch, useSelector } from "react-redux";
import { useEffect } from "react";
import axios from "axios";
import {
    setPlaying,
    toggleLoop,
    toggleRandom,
} from "../../store/actions/playerActions";
import {
    setLoading,
    setQueue,
    setQueueData,
} from "../../store/actions/queueAction";
import {
    setPlayingSong,
    setSongData,
} from "../../store/actions/songDataAction";
import { QueueListComponent } from "./QueueListComponent";

export const QueueContainer = ({ openQueue }) => {
    const apiUrl = process.env.REACT_APP_API_URL;
    const { queue, queueData, loading, loop, random } = useSelector(
        (state) => state.queueReducer
    );
    const playingSong = useSelector(
        (state) => state.songDataReducer.songData.songId
    );
    const dispatch = useDispatch();

    useEffect(() => {
        const getQueueData = async (qdCache) => {
            var data = [];
            dispatch(setLoading(true));

            for (let i = 0; i < queue.length; i++) {
                let found = false;
                for (let j = 0; j < qdCache.length; j++) {
                    if (queue[i] === qdCache[j].songId) {
                        data.push(qdCache[j]);
                        found = true;
                        break;
                    }
                }
                if (!found) {
                    await axios
                        .get(`${apiUrl}/song/${queue[i]}`)
                        .then((result) => {
                            data.push({
                                songId: result.data.songs._id,
                                title: result.data.songs.title,
                                duration: result.data.songs.duration,
                                albumname: result.data.albumname,
                                albumId: result.data._id,
                                artist: result.data.artist,
                            });
                        })
                        .catch((e) => console.log(e));
                }
            }
            return data;
        };

        getQueueData(queueData).then((res) => {
            dispatch(setQueueData(res));
            dispatch(setLoading(false));
            if (res.length <= 0) return;
            if (playingSong) {
                dispatch(
                    setSongData(
                        res[
                            res
                                .map((qd) => {
                                    return qd.songId;
                                })
                                .indexOf(playingSong)
                        ]
                    )
                );
            }
        });
    }, [queue]); // eslint-disable-line

    useEffect(() => {
        if (!playingSong) return;
        const qdList = queueData.map((qd) => {
            return qd.songId;
        });
        if (qdList.includes(playingSong))
            dispatch(setSongData(queueData[qdList.indexOf(playingSong)]));

        const item = document.getElementById(playingSong);
        if (!item) return;
        item.scrollIntoView(false);
    }, [playingSong]); // eslint-disable-line

    const skipQueue = (e) => {
        // If the item is clicked, and it is not the trash can icon, play the clicked song instead
        let clickedItem = e.currentTarget.getAttribute("datakey");
        if (!queue.includes(clickedItem)) return;
        dispatch(setPlayingSong(clickedItem));
        dispatch(setPlaying(true));
    };

    const removeQueue = async (e) => {
        // remove the selected song from queue
        let clickedItem = e.currentTarget.getAttribute("datakey");
        if (clickedItem === playingSong) {
            dispatch(setPlayingSong(""));
            dispatch(setPlaying(false));
        }

        let q = queue;
        q.splice(queue.indexOf(clickedItem), 1);
        dispatch(setQueue(q));
    };

    const handleDragDrop = (result) => {
        // re-position the queue according to the drop location of the dragged item
        if (!result.destination) return;

        const songs = queue;
        const [reorderedSong] = songs.splice(result.source.index, 1);
        songs.splice(result.destination.index, 0, reorderedSong);
        dispatch(setQueue(songs));
    };

    const changeRandom = () => {
        dispatch(toggleRandom());
    };

    const changeLoop = () => {
        dispatch(toggleLoop());
    };

    return (
        <QueueComponent
            isLoop={loop}
            isRandom={random}
            loading={loading}
            toggleRandom={changeRandom}
            toggleLoop={changeLoop}
            handleDragDrop={handleDragDrop}
            queueList={QueueListComponent({
                data: queueData.length > 0 ? queueData : [],
                playingSong,
                skipQueue,
                removeQueue,
            })}
            openQueue={openQueue}
        />
    );
};
