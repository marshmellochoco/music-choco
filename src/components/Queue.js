import Icon from "@mdi/react";
import { useEffect } from "react";
import { mdiClose, mdiShuffle, mdiSync } from "@mdi/js";
import { useDispatch, useSelector } from "react-redux";
import { useLocation } from "react-router";
import { setPlaying, setPlayingTrack } from "../store/player/playerAction";
import { setQueue } from "../store/queue/queueAction";
import { DragDropContext, Draggable, Droppable } from "react-beautiful-dnd";
import { Link } from "react-router-dom/cjs/react-router-dom.min";
import { setLoop } from "../store/player/playerAction";
import { ContextMenu, ContextMenuTrigger, MenuItem } from "react-contextmenu";

const Queue = ({ openQueue, setOpenQueue }) => {
    const location = useLocation();

    useEffect(() => {
        setOpenQueue(false);
        // eslint-disable-next-line
    }, [location]);

    const dispatch = useDispatch();
    const queueData = useSelector((state) => state.queueReducer);
    const { playingTrack, loop } = useSelector((state) => state.playerReducer);

    const removeFromQueue = (q) => {
        let index = queueData.indexOf(q);
        if (
            index === queueData.indexOf(playingTrack) &&
            queueData.length !== 0
        ) {
            if (index === queueData.length - 1)
                dispatch(setPlayingTrack(queueData[index - 1]));
            else dispatch(setPlayingTrack(queueData[index + 1]));
        }
        let newQueue = queueData.filter((_, i) => i !== index);
        dispatch(setQueue(newQueue));
    };

    const handleDragDrop = (result) => {
        if (!result.destination) return;

        const q = [...queueData];
        const [s] = q.splice(result.source.index, 1);
        q.splice(result.destination.index, 0, s);
        dispatch(setQueue(q));
    };

    const setTrack = (trackData) => {
        dispatch(setPlayingTrack(trackData));
        dispatch(setPlaying(true));
    };

    const closeQueue = () => {
        setOpenQueue(false);
    };

    const shuffle = () => {
        let q = [...queueData];
        q.splice(q.indexOf(playingTrack), 1);
        let shuffled = [...q].sort(() => Math.random() - 0.5);
        dispatch(setQueue([playingTrack, ...shuffled]));
    };

    const getQueueList = (queue, playingTrack) => {
        return queue.map((q, i) => {
            const isPlayingTrack = playingTrack._id === q._id;
            return (
                <div key={q._id}>
                    <ContextMenuTrigger id={`queueListContextMenu_${q._id}`}>
                        <Draggable key={q._id} draggableId={q._id} index={i}>
                            {(provided, snapshot) => (
                                <div
                                    onClick={(e) => {
                                        if (
                                            !e.target.className.includes(
                                                "hover:underline"
                                            )
                                        )
                                            setTrack(q);
                                    }}
                                    id={q._id}
                                    className={`${
                                        isPlayingTrack
                                            ? "queue-item-active hover:bg-red-200"
                                            : snapshot.isDragging
                                            ? "queue-item-dragging"
                                            : "hover:bg-red-50"
                                    }`}
                                    ref={provided.innerRef}
                                    {...provided.draggableProps}
                                    {...provided.dragHandleProps}
                                    datakey={q._id}
                                >
                                    <li className="queue-item">
                                        <div className="col-span-6">
                                            <b>{q.title}</b>
                                            <div className="artistList">
                                                {q.artists.map((a) => (
                                                    <Link
                                                        key={a._id}
                                                        to={`/artist/${a._id}`}
                                                        className="hover:underline linkItem"
                                                    >
                                                        {a.name}
                                                    </Link>
                                                ))}
                                            </div>
                                        </div>
                                        <div className="col-span-3 mr-2 artistList">
                                            <Link
                                                to={`/album/${q.album._id}`}
                                                className="hover:underline"
                                            >
                                                {q.album.name}
                                            </Link>
                                        </div>
                                        <div className="text-right">
                                            {(q.duration / 60 < 10 ? "0" : "") +
                                                Math.floor(q.duration / 60)}
                                            :
                                            {(q.duration % 60 < 10 ? "0" : "") +
                                                Math.floor(q.duration % 60)}
                                        </div>
                                    </li>
                                </div>
                            )}
                        </Draggable>
                    </ContextMenuTrigger>
                    <ContextMenu
                        key={q._id + "test"}
                        id={`queueListContextMenu_${q._id}`}
                        className="contextMenu"
                    >
                        <MenuItem
                            onClick={() => removeFromQueue(q)}
                            className="menuItem"
                        >
                            Remove from queue
                        </MenuItem>
                    </ContextMenu>
                </div>
            );
        });
    };

    return (
        <>
            {openQueue && (
                <div className="bg-white content mx-2">
                    <div className="flex justify-between items-center">
                        <h1 className="title">Up Next</h1>
                        <div className="flex gap-10">
                            <div className="flex gap-6">
                                <div
                                    className={`rounded-full hover:bg-red-100 ${
                                        loop
                                            ? "bg-pink-100 hover:bg-red-200"
                                            : ""
                                    }`}
                                >
                                    <Icon
                                        path={mdiSync}
                                        title="Loop"
                                        className="icon-small fill-current text-pink-600"
                                        onClick={() => dispatch(setLoop(!loop))}
                                    />
                                </div>
                                <div className="rounded-full hover:bg-red-100">
                                    <Icon
                                        path={mdiShuffle}
                                        title="Shuffle"
                                        className={
                                            "icon-small fill-current text-pink-600"
                                        }
                                        onClick={shuffle}
                                    />
                                </div>
                            </div>
                            <div className="rounded-full hover:bg-gray-200">
                                <Icon
                                    path={mdiClose}
                                    title="Close"
                                    className="icon-small"
                                    onClick={closeQueue}
                                />
                            </div>
                        </div>
                    </div>
                    <DragDropContext onDragEnd={handleDragDrop}>
                        <Droppable droppableId="droppable">
                            {(provided) => (
                                <ul
                                    {...provided.droppableProps}
                                    ref={provided.innerRef}
                                >
                                    {getQueueList(queueData, playingTrack)}
                                    {provided.placeholder}
                                </ul>
                            )}
                        </Droppable>
                    </DragDropContext>
                </div>
            )}
        </>
    );
};

export default Queue;
