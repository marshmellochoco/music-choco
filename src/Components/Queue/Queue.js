import axios from "axios";
import { useCallback, useEffect, useState } from "react";
import { DragDropContext, Draggable, Droppable } from "react-beautiful-dnd";
import "./Queue.css";

export const Queue = ({
    index,
    setIndex,
    queue,
    setQueue,
    setPlaying,
    apiUrl,
}) => {
    const [queueData, setQueueData] = useState([]);

    const getQueueData = useCallback(async () => {
        var data = [];
        for (let i = 0; i < queue.length; i++) {
            let dataContainer;
            await axios.get(apiUrl + "/song/" + queue[i]).then((res) => {
                dataContainer = res.data;
            });

            await axios
                .get(apiUrl + "/album/" + dataContainer.album)
                .then((res) => {
                    let name = res.data.albumname;
                    data.push({
                        _id: dataContainer._id,
                        title: dataContainer.title,
                        duration: dataContainer.duration,
                        albumname: name,
                    });
                });
        }
        setQueueData(data);
    }, [queue, apiUrl]);

    useEffect(() => {
        getQueueData();
    }, [getQueueData]);

    const skipQueue = (e) => {
        let clickedItem = e.currentTarget.getAttribute("datakey");
        setIndex(queue.indexOf(clickedItem));
        setPlaying(true);
    };

    const removeQueue = (e) => {
        let clickedItem = e.currentTarget.getAttribute("datakey");
        let q = queue;
        q.splice(q.indexOf(clickedItem), 1);
        setQueue([...q]);
    };

    const handleDragDrop = (result) => {
        const data = queueData;
        const [reorderedData] = data.splice(result.source.index, 1);
        data.splice(result.destination.index, 0, reorderedData);
        setQueueData(data);

        const songs = queue;
        const [reorderedSong] = songs.splice(result.source.index, 1);
        songs.splice(result.destination.index, 0, reorderedSong);
        setQueue(songs);
    }

    const getQueueList = (provided) => {
        return queueData.map((q, i) => {
            return (
                <Draggable key={q._id} draggableId={q._id} index={i}>
                    {(provided) => (
                        <li
                            className={`${
                                queue[index] === q._id
                                    ? "queueActive"
                                    : "queueItem"
                            }`}
                            ref={provided.innerRef}
                            {...provided.draggableProps}
                            {...provided.dragHandleProps}
                            datakey={q._id}
                            onClick={(e) => skipQueue(e)}
                            onContextMenu={(e) => removeQueue(e)}
                        >
                            <div>{q.title}</div>
                            <div className="album">{q.albumname}</div>
                            <div className="duration">
                                {" "}
                                {(q.duration / 60 < 10 ? "0" : "") +
                                    Math.floor(q.duration / 60)}{" "}
                                :{" "}
                                {(q.duration % 60 < 10 ? "0" : "") +
                                    Math.floor(q.duration % 60)}
                            </div>
                        </li>
                    )}
                </Draggable>
            );
        });
    };

    return (
        <div className="queue">
            <h1>Queue</h1>
            <DragDropContext onDragEnd={handleDragDrop}>
                <Droppable droppableId="droppable">
                    {(provided) => (
                        <ul
                            className="queueList"
                            {...provided.droppableProps}
                            ref={provided.innerRef}
                        >
                            {getQueueList(provided)}
                            {provided.placeholder}
                        </ul>
                    )}
                </Droppable>
            </DragDropContext>
        </div>
    );
};
