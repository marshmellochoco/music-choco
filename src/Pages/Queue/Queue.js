import axios from "axios";
import React from "react";
import { useEffect, useState } from "react";
import { DragDropContext, Draggable, Droppable } from "react-beautiful-dnd";
import ReactLoading from "react-loading";
import "./Queue.css";

export const Queue = ({
    playingSong,
    setPlayingSong,
    queue,
    setQueue,
    setPlaying,
    apiUrl,
    isRandom,
    setRandom,
    isLoop,
    setLoop,
    setRandomQueue,
}) => {
    const [queueData, setQueueData] = useState([]);
    const [loading, setLoading] = useState(false);

    useEffect(() => {
        const getQueueData = async () => {
            var data = [];
            setQueueData([]);
            setLoading(true);

            for (let i = 0; i < queue.length; i++) {
                let found = false;
                for (let j = 0; j < qdCache.length; j++) {
                    if (queue[i] === qdCache[j]._id) {
                        data.push(qdCache[j]);
                        found = true;
                        break;
                    }
                }

                if (!found) {
                    await axios
                        .get(apiUrl + "/song/" + queue[i])
                        .then((result) => {
                            data.push({
                                _id: result.data.songs._id,
                                title: result.data.songs.title,
                                duration: result.data.songs.duration,
                                albumname: result.data.albumname
                            })
                        });
                }
            }
            setQueueData(data);
            setLoading(false);
        };

        const qdCache = queueData;
        getQueueData();

        let randQueue = [...queue];
        setRandomQueue(randQueue.sort(() => Math.random() - 0.5));
    }, [queue]); // eslint-disable-line

    useEffect(() => {
        if (!playingSong) return;
        const item = document.getElementById(playingSong);
        if (!item) return;
        item.scrollIntoView(false);
    }, [playingSong]);

    const skipQueue = (e) => {
        if (e.target.localName !== "svg") {
            let clickedItem = e.currentTarget.getAttribute("datakey");
            setPlayingSong(clickedItem);
            setPlaying(true);
        }
    };

    const removeQueue = async (e) => {
        let clickedItem = e.currentTarget.getAttribute("datakey");

        let q = queue;
        q.splice(queue.indexOf(clickedItem), 1);
        setQueue([...q]);

        if (clickedItem === playingSong) {
            setPlayingSong("");
            setPlaying(false);
        }
    };

    const handleDragDrop = (result) => {
        if (!result.destination) return;
        const data = queueData;
        const [reorderedData] = data.splice(result.source.index, 1);
        data.splice(result.destination.index, 0, reorderedData);
        setQueueData(data);

        const songs = queue;
        const [reorderedSong] = songs.splice(result.source.index, 1);
        songs.splice(result.destination.index, 0, reorderedSong);
        setQueue(songs);
    };

    const getQueueList = (provided, data) => {
        return data.map((q, i) => {
            return (
                <Draggable key={q._id} draggableId={q._id} index={i}>
                    {(provided, snapshot) => (
                        <div
                            id={q._id}
                            className={`${
                                playingSong === q._id
                                    ? "queueActive"
                                    : "queueItem"
                            }
                            ${snapshot.isDragging ? "dragging" : ""}`}
                            ref={provided.innerRef}
                            {...provided.draggableProps}
                            {...provided.dragHandleProps}
                            datakey={q._id}
                            onClick={(e) => skipQueue(e)}
                            onContextMenu={(e) => {
                                removeQueue(e);
                            }}
                        >
                            <li>
                                <div className="title">{q.title}</div>
                                <div className="album">{q.albumname}</div>
                                <div
                                    className={"removeButton icon"}
                                    datakey={q._id}
                                    onClick={(e) => removeQueue(e)}
                                >
                                    <svg viewBox="0 0 24 24">
                                        <path
                                            fill="currentColor"
                                            d="M9,3V4H4V6H5V19A2,2 0 0,0 7,21H17A2,2 0 0,0 19,19V6H20V4H15V3H9M7,6H17V19H7V6M9,8V17H11V8H9M13,8V17H15V8H13Z"
                                        />
                                    </svg>
                                </div>
                                <div className="duration">
                                    {" "}
                                    {(q.duration / 60 < 10 ? "0" : "") +
                                        Math.floor(q.duration / 60)}{" "}
                                    :{" "}
                                    {(q.duration % 60 < 10 ? "0" : "") +
                                        Math.floor(q.duration % 60)}
                                </div>
                            </li>
                        </div>
                    )}
                </Draggable>
            );
        });
    };

    return (
        <div className="queue">
            <div className="queueHeader">
                <h1>Queue</h1>
                <div className="controls">
                    <svg
                        xmlns="http://www.w3.org/2000/svg"
                        height="24px"
                        viewBox="0 0 24 24"
                        width="24px"
                        className={`icons ${isRandom ? "active" : ""}`}
                        onClick={() => setRandom(!isRandom)}
                    >
                        <path d="M10.59 9.17L5.41 4 4 5.41l5.17 5.17 1.42-1.41zM14.5 4l2.04 2.04L4 18.59 5.41 20 17.96 7.46 20 9.5V4h-5.5zm.33 9.41l-1.41 1.41 3.13 3.13L14.5 20H20v-5.5l-2.04 2.04-3.13-3.13z" />
                    </svg>
                    <svg
                        xmlns="http://www.w3.org/2000/svg"
                        height="24px"
                        viewBox="0 0 24 24"
                        width="24px"
                        className={`icons ${isLoop ? "active" : ""}`}
                        onClick={() => setLoop(!isLoop)}
                    >
                        <path d="M12 4V1L8 5l4 4V6c3.31 0 6 2.69 6 6 0 1.01-.25 1.97-.7 2.8l1.46 1.46C19.54 15.03 20 13.57 20 12c0-4.42-3.58-8-8-8zm0 14c-3.31 0-6-2.69-6-6 0-1.01.25-1.97.7-2.8L5.24 7.74C4.46 8.97 4 10.43 4 12c0 4.42 3.58 8 8 8v3l4-4-4-4v3z" />
                    </svg>
                </div>
            </div>
            {loading ? (
                <ReactLoading className="loading" type="bars" />
            ) : (
                <DragDropContext onDragEnd={handleDragDrop}>
                    <Droppable droppableId="droppable">
                        {(provided) => (
                            <ul
                                className="queueList"
                                {...provided.droppableProps}
                                ref={provided.innerRef}
                            >
                                {getQueueList(provided, queueData)}
                                {provided.placeholder}
                            </ul>
                        )}
                    </Droppable>
                </DragDropContext>
            )}
        </div>
    );
};
