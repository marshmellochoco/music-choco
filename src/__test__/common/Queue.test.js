import { configure, mount } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import toJson from "enzyme-to-json";
import { Provider } from "react-redux";
import { createStore } from "redux";

import { rootReducer } from "../../store/rootReducer";
import { QueueContainer } from "../../common/Queue/QueueContainer";

configure({ adapter: new Adapter() });
const store = createStore(rootReducer);

describe("Queue", () => {
    it("should render the queue", () => {
        const wrapper = mount(
            <Provider store={store}>
                <QueueContainer />
            </Provider>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
