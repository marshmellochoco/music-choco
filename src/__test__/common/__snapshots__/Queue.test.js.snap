// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queue should render the queue 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <QueueContainer>
    <QueueComponent
      handleDragDrop={[Function]}
      loading={true}
      queueList={Array []}
      toggleLoop={[Function]}
      toggleRandom={[Function]}
    >
      <EmotionCssPropInternal
        __EMOTION_LABEL_PLEASE_DO_NOT_USE__="QueueComponent"
        __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
        css={
          Object {
            "map": undefined,
            "name": "15olhuf",
            "next": undefined,
            "styles": "
        background-color: var(--secondary-color);
        padding: 0 1rem;
        overflow-y: scroll;
        display: none;

        & > .loading {
            margin: auto;
        }

        @media (max-width: 1024px) {
            width: calc(100% - 3rem);
            height: 100%;
            margin-right: 1rem;
            background-color: var(--secondary-color);
        }
    ",
            "toString": [Function],
          }
        }
      >
        <div
          className="css-epg54b-QueueComponent"
        >
          <EmotionCssPropInternal
            __EMOTION_LABEL_PLEASE_DO_NOT_USE__="QueueComponent"
            __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
            css={
              Object {
                "map": undefined,
                "name": "1oagz98",
                "next": undefined,
                "styles": "
        display: flex;
        align-items: baseline;
        align-content: flex-end;
        margin-left: auto;

        & > div {
            display: flex;
            margin-left: auto;

            & > svg {
                fill: var(--contrast-color);
                width: 2em;
                height: 2em;
                margin: 1rem;
                cursor: pointer;

                &:hover {
                    opacity: 0.7;
                }
            }

            & > .active {
                color: var(--primary-color);
            }
        }
    ",
                "toString": [Function],
              }
            }
          >
            <div
              className="css-95yp7n-QueueComponent"
            >
              <h1>
                Queue
              </h1>
              <div>
                <Icon
                  className=""
                  color="currentColor"
                  horizontal={false}
                  onClick={[Function]}
                  path="M14.83,13.41L13.42,14.82L16.55,17.95L14.5,20H20V14.5L17.96,16.54L14.83,13.41M14.5,4L16.54,6.04L4,18.59L5.41,20L17.96,7.46L20,9.5V4M10.59,9.17L5.41,4L4,5.41L9.17,10.58L10.59,9.17Z"
                  rotate={0}
                  size={null}
                  spin={false}
                  vertical={false}
                >
                  <svg
                    className=""
                    onClick={[Function]}
                    role="presentation"
                    style={Object {}}
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M14.83,13.41L13.42,14.82L16.55,17.95L14.5,20H20V14.5L17.96,16.54L14.83,13.41M14.5,4L16.54,6.04L4,18.59L5.41,20L17.96,7.46L20,9.5V4M10.59,9.17L5.41,4L4,5.41L9.17,10.58L10.59,9.17Z"
                      style={
                        Object {
                          "fill": "currentColor",
                        }
                      }
                    />
                  </svg>
                </Icon>
                <Icon
                  className=""
                  color="currentColor"
                  horizontal={false}
                  onClick={[Function]}
                  path="M12,18A6,6 0 0,1 6,12C6,11 6.25,10.03 6.7,9.2L5.24,7.74C4.46,8.97 4,10.43 4,12A8,8 0 0,0 12,20V23L16,19L12,15M12,4V1L8,5L12,9V6A6,6 0 0,1 18,12C18,13 17.75,13.97 17.3,14.8L18.76,16.26C19.54,15.03 20,13.57 20,12A8,8 0 0,0 12,4Z"
                  rotate={0}
                  size={null}
                  spin={false}
                  vertical={false}
                >
                  <svg
                    className=""
                    onClick={[Function]}
                    role="presentation"
                    style={Object {}}
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12,18A6,6 0 0,1 6,12C6,11 6.25,10.03 6.7,9.2L5.24,7.74C4.46,8.97 4,10.43 4,12A8,8 0 0,0 12,20V23L16,19L12,15M12,4V1L8,5L12,9V6A6,6 0 0,1 18,12C18,13 17.75,13.97 17.3,14.8L18.76,16.26C19.54,15.03 20,13.57 20,12A8,8 0 0,0 12,4Z"
                      style={
                        Object {
                          "fill": "currentColor",
                        }
                      }
                    />
                  </svg>
                </Icon>
              </div>
            </div>
          </EmotionCssPropInternal>
          <Loading
            className="loading"
            color="#fff"
            delay={0}
            height={64}
            type="bars"
            width={64}
          >
            <div
              className="loading"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 32 32\\">
  <path transform=\\"translate(2)\\" d=\\"M0 12 V20 H4 V12z\\"> 
    <animate attributeName=\\"d\\" values=\\"M0 12 V20 H4 V12z; M0 4 V28 H4 V4z; M0 12 V20 H4 V12z; M0 12 V20 H4 V12z\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0\\" keytimes=\\"0;.2;.5;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.8 0.4 0.8\\" calcMode=\\"spline\\"  />
  </path>
  <path transform=\\"translate(8)\\" d=\\"M0 12 V20 H4 V12z\\">
    <animate attributeName=\\"d\\" values=\\"M0 12 V20 H4 V12z; M0 4 V28 H4 V4z; M0 12 V20 H4 V12z; M0 12 V20 H4 V12z\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.2\\" keytimes=\\"0;.2;.5;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.8 0.4 0.8\\" calcMode=\\"spline\\"  />
  </path>
  <path transform=\\"translate(14)\\" d=\\"M0 12 V20 H4 V12z\\">
    <animate attributeName=\\"d\\" values=\\"M0 12 V20 H4 V12z; M0 4 V28 H4 V4z; M0 12 V20 H4 V12z; M0 12 V20 H4 V12z\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.4\\" keytimes=\\"0;.2;.5;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.8 0.4 0.8\\" calcMode=\\"spline\\" />
  </path>
  <path transform=\\"translate(20)\\" d=\\"M0 12 V20 H4 V12z\\">
    <animate attributeName=\\"d\\" values=\\"M0 12 V20 H4 V12z; M0 4 V28 H4 V4z; M0 12 V20 H4 V12z; M0 12 V20 H4 V12z\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.6\\" keytimes=\\"0;.2;.5;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.8 0.4 0.8\\" calcMode=\\"spline\\" />
  </path>
  <path transform=\\"translate(26)\\" d=\\"M0 12 V20 H4 V12z\\">
    <animate attributeName=\\"d\\" values=\\"M0 12 V20 H4 V12z; M0 4 V28 H4 V4z; M0 12 V20 H4 V12z; M0 12 V20 H4 V12z\\" dur=\\"1.2s\\" repeatCount=\\"indefinite\\" begin=\\"0.8\\" keytimes=\\"0;.2;.5;1\\" keySplines=\\"0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.8 0.4 0.8\\" calcMode=\\"spline\\" />
  </path>
</svg>
",
                }
              }
              style={
                Object {
                  "fill": "#fff",
                  "height": 64,
                  "width": 64,
                }
              }
            />
          </Loading>
        </div>
      </EmotionCssPropInternal>
    </QueueComponent>
  </QueueContainer>
</Provider>
`;
